test/unit/mdc-top-app-bar/foundation.test.js

```diff
@@ -58,3 +77,68 @@ test('click handler removed from navigation icon during destroy', () => {
   foundation.destroy();
   td.verify(mockAdapter.deregisterNavigationIconInteractionHandler('click', td.matchers.isA(Function)));
 });
+
+test('short top app bar: scroll listener is registered on init', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+  foundation.init();
+  td.verify(mockAdapter.registerScrollHandler(td.matchers.isA(Function)), {times: 1});
+});
+
+test('short top app bar: scroll listener is removed on destroy', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+  foundation.init();
+  foundation.destroy();
+  td.verify(mockAdapter.registerScrollHandler(td.matchers.isA(Function)), {times: 1});
+});
+
+test('short top app bar: class is added once when page is scrolled from the top', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const mockRaf = createMockRaf();
+
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+
+  const {scrollHandler} = createMockHandlers(foundation, mockAdapter, mockRaf);
+
+  td.when(mockAdapter.getViewportScrollY()).thenReturn(1);
+  scrollHandler();
+  scrollHandler();
+
+  td.verify(mockAdapter.addClass(MDCTopAppBarFoundation.cssClasses.SHORT_COLLAPSED_CLASS), {times: 1});
+});
+
+test('short top app bar: class is removed once when page is scrolled to the top', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const mockRaf = createMockRaf();
+
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+
+  const {scrollHandler} = createMockHandlers(foundation, mockAdapter, mockRaf);
+  // Apply the closed class
+  td.when(mockAdapter.getViewportScrollY()).thenReturn(1);
+  scrollHandler();
+
+  // Test removing it
+  td.when(mockAdapter.getViewportScrollY()).thenReturn(0);
+  scrollHandler();
+  scrollHandler();
+
+  td.verify(mockAdapter.removeClass(MDCTopAppBarFoundation.cssClasses.SHORT_COLLAPSED_CLASS), {times: 1});
+});
+
+test('short top app bar: class is added if it has an action item', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getTotalActionItems()).thenReturn(1);
+  foundation.init();
+  td.verify(mockAdapter.addClass(MDCTopAppBarFoundation.cssClasses.SHORT_HAS_ACTION_ITEM_CLASS), {times: 1});
+});
+
+test('short top app bar: class is not added if it does not have an action item', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getTotalActionItems()).thenReturn(0);
+  foundation.init();
+  td.verify(mockAdapter.addClass(MDCTopAppBarFoundation.cssClasses.SHORT_HAS_ACTION_ITEM_CLASS), {times: 0});
```

Comment:
Our impl is actually removing this class in this case, right? We should test that removeClass is invoked.

Reply:
We're testing that this class was never added. It's added during init if the html has an action item.